src/omuse/community/pop/interface.py

```diff
@@ -1,1213 +1,1655 @@
-        object.add_transition('EDIT', 'RUN', 'recommit_grids')
-        object.add_transition('EDIT_FORCINGS', 'RUN', 'recommit_forcings')
-        object.add_transition('EDIT_FORCINGS', 'EDIT', 'recommit_forcings')
-        object.add_transition('EDIT', 'EDIT_FORCINGS', 'prepare_forcings') # prepare_forcings is a dummy
-        object.add_transition('RUN', 'EDIT_FORCINGS', 'prepare_forcings')
-        object.add_transition('RUN', 'EDIT', 'prepare_forcings')
-        
-
-        #can only evolve from run
-        object.add_transition('RUN', 'EVOLVED', 'evolve_model', False)
-
-        #from state EVOLVED you may evolve again if you like
-        object.add_method('EVOLVED','evolve_model')
-
-        #prepare model for reading/writing after evolve
-        object.add_transition('EVOLVED', 'RUN', 'prepare_parameters')
-        object.add_transition('EVOLVED', 'EDIT', 'prepare_parameters')
-
-        #ensure that get_model_time only returns valid values
-        object.add_method('RUN', 'get_model_time')
-        object.add_method('EDIT', 'get_model_time')
-        object.add_method('EDIT_FORCINGS', 'get_model_time')
-        object.add_method('EVOLVED', 'get_model_time')
+        object.set_initial_state("UNINITIALIZED")
+        object.add_method("UNINITIALIZED", "change_directory")
+        object.add_method("UNINITIALIZED", "set_namelist_filename")
+
+        object.add_transition(
+            "!UNINITIALIZED!INITIALIZED!STOPPED", "END", "cleanup_code"
+        )
+        object.add_transition("UNINITIALIZED", "INITIALIZED", "initialize_code")
+        object.add_transition("END", "STOPPED", "stop", False)
+        object.add_method("STOPPED", "stop")
+
+        # some parameters have to be derived before they can be read, therefore
+        # reading anything should be preceded by a call to prepare_parameters
+        # in the two-phase init, prepare_parameters is called by commit_parameters()
+        # commit_parameters must be called to finish the initialization
+        object.add_transition("INITIALIZED", "EDIT", "commit_parameters")
+        object.add_transition("INITIALIZED", "RUN", "commit_parameters")
+
+        object.add_method("INITIALIZED", "set_horiz_grid_option")
+        object.add_method("INITIALIZED", "set_horiz_grid_file")
+        object.add_method("INITIALIZED", "set_vert_grid_option")
+        object.add_method("INITIALIZED", "set_vert_grid_file")
+        object.add_method("INITIALIZED", "set_topography_option")
+        object.add_method("INITIALIZED", "set_topography_file")
+
+        object.add_method("INITIALIZED", "set_shf_data_type")
+        object.add_method("INITIALIZED", "set_shf_monthly_file")
+        object.add_method("INITIALIZED", "set_sfwf_data_type")
+        object.add_method("INITIALIZED", "set_sfwf_monthly_file")
+        object.add_method("INITIALIZED", "set_ws_monthly_file")
+
+        # ~ object.add_method('INITIALIZED', 'get_ts_option')
+        object.add_method("INITIALIZED", "set_ts_option")
```

Comment:
It seems to work after I remove this line. Note sure if that still does the right thing? I don't see any obvious additional errors in the solution.

Reply:
I think this is indeed fine to remove this constrain.

Reply:
Are you trying to call `set_ts_option` on an uninitialized object or an object that has already ran/evolved ? Removing this entirely means that this could be called on an uninitialized object, but it will be overwritten during the POP initialized. So we can add the method to the `RUN` or `EVOLVED` state instead of removing this line.

Reply:
One where parameters were already committed. So I think it's in the `EDIT` state. If you change anything here, I imagine you want to do something similar for the other parameters like `reinit_gradp`.